src/main/java/com/mathworks/ci/SelectByFolder.java

```diff
@@ -0,0 +1,38 @@
+package com.mathworks.ci;
+
+import java.util.List;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+
+public class SelectByFolder extends AbstractDescribableImpl<SelectByFolder> {
+    private List<TestFolders> testFolderPaths;
+    private static final String SELECT_BY_FOLDER = "SelectByFolder";
+
+    @DataBoundConstructor
+    public SelectByFolder(List<TestFolders> testFolderPaths) {
```

Comment:
Nikhil, you probably missed adding 'SelectByFolder/config.jelly' file that has describable class SelectByFolder.java  and sets this list of 'testFolderPaths'

Probably HTML tests were failing because of that? As the optional property in RunMatlabTestsBuilder/config.jelly expects another jelly file with the name selectByFolder to be present to render elements when the checkbox is clicked.

Reply:
Thanks @abhargav-mw I realized that now . Thanks for pointing it out. Hopefully, Source Folder test does not have any issue with that :)